### DISCUSSION TITLE
Unable to install vite_rails with rails minimal installation

### DISCUSSION THREAD

**Discussion by ekremkaraca** [ElMassimo/vite_ruby#72] [Q&A] [answered]
Hey guys,

I've created new rails app with minimal flag like `rails new theapp --minimal`, then changed current directory into it. Since it has no `package.json`, I've initiated it with `yarn init -y` command. Then I've added `vite_rails` with `bundle add vite_rails`. Next, I've perfomed `bundle exec vite install` command and finally I get the output below:

```shell
❯ bundle exec vite install
Creating binstub
Check that your vite.json configuration file is available in the load path:

	No such file or directory @ rb_sysopen - /tmp/theapp/config/vite.json

Creating configuration files
Installing sample files
Installing js dependencies
? Choose the agent › - Use arrow-keys. Return to submit.
❯   npm
    yarn
    pnpm


Adding files to .gitignore

Vite ⚡️ Ruby successfully installed! 🎉
```

During `? Choose the agent › - Use arrow-keys. Return to submit.` section, I was unable to choose one of three options since it seems skipped. So, here's `package.json` content that shows no package were installed:

```
{
  "name": "theapp",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT"
}

```

When I hit `bin/vite dev`, I got the error below:

```shell
❯ bin/vite dev
/home/ekrem/.rvm/gems/ruby-3.0.1/gems/vite_ruby-1.2.8/lib/vite_ruby/runner.rb:18:in `rescue in run': ❌ The vite binary is not available. Have you installed Vite? (ViteRuby::MissingExecutableError)

```

If I add `vite` & `vite-plugin-ruby` manually, the application runs with vite output in browser console. Is it possible to install and use `vite_rails` with rails app created with `--minimal` flag or should I initiate rails app as usual, then migrate from webpacker?

**Comment by ElMassimo** [comment, marked as answer]
Hi Ekrem!

During the install command, the library uses `@antfu/ni` in order to infer which package manager you are using.

It should be possible to make that command interactive in the future, at the moment based on your experience it skips installation which is not ideal.

The prompt would be skipped if you already had a `yarn.lock` file (which is always there on a "regular" Rails installation).

A workaround for clean installs would be to create a `yarn.lock` or similar file for your preferred package manager, and run `bundle exec vite install` later.

**Reply by ekremkaraca** [reply]
@ElMassimo yep, creating empty `yarn.lock` file seems preventing the error I've faced before. For now, it works. Thanks!

**Reply by ElMassimo** [reply]
Cool, I'll add "Making the install command interactive" to my wishlist.

Thanks for reporting!
